### DISCUSSION TITLE
Simple PV System

### DISCUSSION THREAD

**Discussion by phuismann** [pvlib/pvlib-python#1512] [Q&A]
I'm trying to model a simple py system with pvlib and could not find a suitable example. 
What I'm currently doing is starting out from GHI (ERA5 - reanalysis weather data) and computing all necessary data for pvwatts:

1 - DNI (irradiance.disc with GHI, zenith [...])
2 - DHI  = GHI - DNI * cos(zenith)
3 - POA_Global (pvsystems.get_irradiance, haydavies) 
4 - watts_dc with pvsystems.pvwatts_dc (with POA_direct)

The pvwatts result is not what I expected and does not fit with measurement data: 

https://imgur.com/a/oh5fSEt

Is there anything immediately visible in my approach that is wrong? 

Complete visualisation of all outputs:

https://imgur.com/a/vs8tNO9

https://imgur.com/a/Wd9pwBM

Code:

```
loc = Location(latitude, longitude, tz='Europe/Berlin', altitude=60, name="")
solarposition = loc.get_solarposition(dt, pressure=pressure, temperature=temperature)

solartable = pd.DataFrame()
solartable.index = dt
solartable["time"] = dt
solartable["G"] = G
solartable["zenith"] = solarposition["zenith"]
solartable["azimuth"] = solarposition["azimuth"]
solartable["temperature"] = temperature - 273.13
solartable["pressure"] = pressure
solartable["v_abs"] = v_abs
solartable["dni"] = irradiance.disc(ghi=solartable["G"],
                                    solar_zenith=solartable['zenith'],
                                    datetime_or_doy=solartable["time"].dt.dayofyear,
                                    pressure=solartable['pressure'])["dni"]

solartable["dhi"] = solartable["G"] - solartable["dni"] * np.cos(np.radians(solartable["zenith"]))

pdc0 = 5039 

module_params = {"pdc0": pdc0, "gamma_pdc": -0.002}

pv = pvlib.pvsystem.PVSystem(surface_tilt=unit["tilt"],
                             surface_azimuth=unit["azi"],
                             module_parameters=module_params)

df_res = pv.get_irradiance(solartable["zenith"], solartable["azimuth"],
                           solartable["dni"], solartable["G"], solartable["dhi"], model="haydavies")

cell_temp = pv.get_cell_temperature(poa_global=df_res["poa_global"],
                                    temp_air=solartable["temperature"],
                                    wind_speed=solartable["v_abs"],
                                    model="pvsyst")

watts_dc = pv.pvwatts_dc(g_poa_effective=df_res["poa_global"], temp_cell=cell_temp)
```

**Comment by AdamRJensen** [comment]
Without seeing the code it's hard to see if you're doing anything wrong. As a first step, I would highly recommend you to visualize your data, e.g., start by looking at only a few days and see if the DNI/GHI/POA/zenith seems reasonable.

**Reply by phuismann** [reply]
Thank you for the fast reply! I have added code and visualisation of most of the time series. For me, poa_direct seems to be off.

**Reply by kandersolar** [reply]
`pv.pvwatts_dc(g_poa_effective=df_res["poa_direct"], ...)` -- should this not use `df_res['poa_global']`?

I second the suggestion of visualizing at the scale of a few days in addition to the full annual time series.  One common issue, especially if `poa_direct` seems wonky, is that the time index is not properly timezone localized.  That would cause the calculated solar positions to be shifted in time with respect to the GHI data and mess up the `poa_direct` calculation. 

Something else to think about if you aren't already is that calculating solar position for the edge of each hourly interval introduces some avoidable error for averaged data like ERA5 irradiance.  See similar discussion here: https://github.com/PyPSA/atlite/issues/158

Otherwise, can you elaborate on "The pvwatts result is not what I expected and does not fit with measurement data" -- specifically how is it not what you expected?

**Reply by phuismann** [reply]
Yes it's poa_global not direct.. I tried something and copied this error. 

The pvwatts result does not fit with measured data (monthly sum more than yearly sum of produced energy) and the shift of highest production to earlier in the year seems odd.

**Reply by phuismann** [reply]
Looking at a weekly result (https://imgur.com/a/uE5RRUd) it looks realistic.
